.github/workflows/python.yml

```diff
@@ -0,0 +1,126 @@
+
+jobs:
+  # Generate the sdist first. We'll use it to created the wheels.
+  # https://packaging.python.org/en/latest/flow#the-source-distribution-sdist
+  sdist:
+    name: Generate sdist
+    runs-on: ubuntu-latest
+    outputs:
+      sdist_filename: ${{ steps.generate.outputs.filename }}
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - uses: actions/setup-python@v4
+      with:
+        python-version: 3.11
+
+    - name: Install build command
+      run: python -m pip install build
+
+    - name: Generate sdist
+      id: generate
+      run: |
+        python -m build -s . --outdir dist
+        echo "filename=$(ls dist)" >> "$GITHUB_OUTPUT"
+
+    - name: Store sdist
+      uses: actions/upload-artifact@v3
+      with:
+        name: sdist
+        path: ./dist/*.tar.gz
+
+  # Create the wheels. It'll use the sdist to confirm that we can compile MaterialX from the sdist.
+  # https://packaging.python.org/en/latest/flow#the-built-distributions-wheels
+  wheels:
+    name: Generate wheels
+    runs-on: ${{ matrix.os }}
+    needs: ['sdist']
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ['37', '38', '39', '310']
+        os: ['ubuntu-latest', 'macos-latest', 'windows-latest']
+
+    steps:
+    - name: Download sdist
+      uses: actions/download-artifact@v3
+      with:
+        name: sdist
+        path: sdist
+
+    - name: Build wheels
+      # https://cibuildwheel.readthedocs.io/en/stable/
+      uses: pypa/cibuildwheel@v2.12.1
+      with:
+        # Build from the sdist. We want to make sure it's valid and works as expected.
+        package-dir: ${{ github.workspace }}/sdist/${{ needs.sdist.outputs.sdist_filename }}
+        output-dir: wheels
+      env:
+        CIBW_BUILD: 'cp${{ matrix.python-version }}-*'
+        CIBW_SKIP: '*musllinux*'
+        CIBW_ARCHS: 'auto64'
+        CIBW_MANYLINUX_X86_64_IMAGE: manylinux2014
+        CIBW_BUILD_VERBOSITY: 1
+        CIBW_ENVIRONMENT: CMAKE_BUILD_PARALLEL_LEVEL=2
+        # CIBW_BUILD_FRONTEND: build  # https://github.com/pypa/build
+        MACOSX_DEPLOYMENT_TARGET: '10.13'
+
+    - name: Store wheels
+      uses: actions/upload-artifact@v3
+      with:
+        name: wheels
+        path: ./wheels/*.whl
+
+  test:
+    name: Test wheels
+    needs: ['wheels']
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ['3.7', '3.8', '3.9', '3.10']
+        os: ['ubuntu-latest', 'macos-latest', 'windows-latest']
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Download wheels
+      uses: actions/download-artifact@v3
+      with:
+        name: wheels
+        path: wheels
+
+    - name: Install wheel
+      run: python -m pip install MaterialX --find-links wheels --no-index
+
+    - name: Python tests
+      run: |
+        python python/MaterialXTest/main.py
```

Comment:
Not a requirement for this PR, but @jstone-lucasfilm ,
seems like it's worthwhile to batch these up so that a single script can be used to run all of these for all CI workflows and for local devs.

Reply:
From what I see, these could all be run from the `main.py` file as "unit tests".